### PR TITLE
policies for IDP management operator

### PR DESCRIPTION
**Pull request by ckandag**
Signed-off-by: ckandag <ckandaga@redhat.com>

https://github.com/open-cluster-management/backlog/issues/17737

### CODE REVIEW

File: community/README.md [hunk 1]
```diff
@@ -123,6 +123,8 @@ Policy  | Description | Prerequisites
...
 [Policy to customize OpenShift OAuth tokens](./CM-Configuration-Management/policy-oauth-config.yaml) | Use this policy to configure the OpenShift tokens to expire after a set period of inactivity. | For more information on configuring the OAuth clients, see the OpenShift documentation: [Configurating the internal oauth Server](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.7/html-single/authentication_and_authorization/index#configuring-internal-oauth)
+[Policy to install IDP operator](./CM-Configuration-Management/policy-idp-operator.yaml) | Use this policy to install Identity configuration management operator. | For more information on this operator, see the IDP documentation: [Identity configuration management for Kubernetes](https://identitatem.github.io/idp-mgmt-docs/)
```
Comment (gparvin) [inline community/README.md]: The policies have comments listing some prereqs, but the prereqs aren't listed in the README.  I didn't look at the link, but stating there are some prereqs in the README would help

File: community/CM-Configuration-Management/policy-idp-operator.yaml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+# my testing ,  some of the config params like the release image, channel/source of the subscription resource
+# will need to change after the final release is cut.
+
+apiVersion: policy.open-cluster-management.io/v1
+kind: Policy
+metadata:
+  annotations:
+    policy.open-cluster-management.io/categories: PR.DS Data Security
+    policy.open-cluster-management.io/controls: PR.DS-1 Data-at-rest
+    policy.open-cluster-management.io/standards: NIST-CSF
```
Comment (ycao56) [inline community/CM-Configuration-Management/policy-idp-operator.yaml]: Can you update the standards/controls/categories to NIST 800-53?

File: community/CM-Configuration-Management/policy-idp-operator.yaml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+spec:
+  disabled: false
+  policy-templates:
+  - objectDefinition:
+      apiVersion: policy.open-cluster-management.io/v1
+      kind: ConfigurationPolicy
+      metadata:
+        name: policy-idp-operator
+      spec:
+        namespaceSelector:
```
Comment (ycao56) [inline community/CM-Configuration-Management/policy-idp-operator.yaml]: `namespaceSelector` is optional. You can remove as the underlying templates are either cluster scope resource or namespace already specified.

File: community/CM-Configuration-Management/policy-idp-sample-github.yaml [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+spec:
+  disabled: false
+  policy-templates:
+  - objectDefinition:
+      apiVersion: policy.open-cluster-management.io/v1
+      kind: ConfigurationPolicy
+      metadata:
+        name: policy-idp-sample-github
+      spec:
+        namespaceSelector:
```
Comment (ycao56) [inline community/CM-Configuration-Management/policy-idp-sample-github.yaml]: same as above

File: community/CM-Configuration-Management/policy-idp-sample-github.yaml [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+# PRE-REQS : The Github OAuth app must have been already created in  Github.
+# the placeholders CLIENTID and CLIENTSECRET in this file must  be replaced with the values from this Github Oauth App
+
+apiVersion: policy.open-cluster-management.io/v1
+kind: Policy
+metadata:
+  annotations:
+    policy.open-cluster-management.io/categories: PR.DS Data Security
+    policy.open-cluster-management.io/controls: PR.DS-1 Data-at-rest
+    policy.open-cluster-management.io/standards: NIST-CSF
```
Comment (ycao56) [inline community/CM-Configuration-Management/policy-idp-sample-github.yaml]: same as above

File: community/CM-Configuration-Management/policy-idp-operator.yaml [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+apiVersion: policy.open-cluster-management.io/v1
+kind: Policy
+metadata:
+  annotations:
+    policy.open-cluster-management.io/categories: PR.DS Data Security
+    policy.open-cluster-management.io/controls: PR.DS-1 Data-at-rest
+    policy.open-cluster-management.io/standards: NIST-CSF
+  name: policy-idp-operator
+  namespace: default
```
Comment (ycao56) [inline community/CM-Configuration-Management/policy-idp-operator.yaml]: namespace should not be hard coded in the policy.

File: community/CM-Configuration-Management/policy-idp-sample-github.yaml [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+  name: policy-idp-sample-github
+  namespace: default
```
Comment (ycao56) [inline community/CM-Configuration-Management/policy-idp-sample-github.yaml]: same as above

**Review by ycao56 — CHANGES_REQUESTED** [review]

File: community/README.md (hunk 1 above)
Comment (ckandag) [reply community/README.md]: ok ty, added pre-reqs to the readme section

File: community/CM-Configuration-Management/policy-idp-operator.yaml (hunk 3 above)
Comment (ckandag) [reply community/CM-Configuration-Management/policy-idp-operator.yaml]: fixed

File: community/CM-Configuration-Management/policy-idp-sample-github.yaml (hunk 4 above)
Comment (ckandag) [reply community/CM-Configuration-Management/policy-idp-sample-github.yaml]: fixed

File: community/CM-Configuration-Management/policy-idp-operator.yaml (hunk 6 above)
Comment (ckandag) [reply community/CM-Configuration-Management/policy-idp-operator.yaml]: fixed

File: community/CM-Configuration-Management/policy-idp-sample-github.yaml (hunk 5 above)
Comment (ckandag) [reply community/CM-Configuration-Management/policy-idp-sample-github.yaml]: fixed

File: community/CM-Configuration-Management/policy-idp-sample-github.yaml (hunk 7 above)
Comment (ckandag) [reply community/CM-Configuration-Management/policy-idp-sample-github.yaml]: fixed

**Review by ycao56 — APPROVED** [review]

**Pull request merged**